pkg/envoy/registry/debugger.go

```diff
@@ -0,0 +1,55 @@
+package registry
+
+import (
+	"time"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/envoy"
+)
+
+// ListExpectedProxies lists the Envoy proxies yet to connect and the time their XDS certificate was issued.
+func (pr *ProxyRegistry) ListExpectedProxies() map[certificate.CommonName]time.Time {
```

Comment:
see https://github.com/openservicemesh/osm/issues/3105#issuecomment-817084945

Reply:
Would it make sense to address those optimizations later, and keep this PR free of functional changes?

Reply:
I... think it's part of the problem. If these changes hurt later on the ability to do efficient data lookups, it has to be accounted beforehand. I'm not saying this commit has to address it, but at least the reason for the change (or the issue) should formalize an answer to that problem.

I'm just trying to prevent us running on dead ends (if actually) in the long run before we even turn on that street.

Reply:
@eduser25 the comment you point to makes sense.
I do thing that we can work on this as a follow-up step.
We call this a ProxyRegistry, and we stop short of reaching for ServiceRegistry. But we could get there and I think this PR here is a step in this direction.

Reply:
@eduser25 Could you open up a new issue for this so it doesn't get lost? Then we can close out #3105 when this is merged.